sys/net/gnrc/link_layer/csma_sender/gnrc_csma_sender.c

```diff
@@ -0,0 +1,245 @@
+    }
+
+    return period;
+}
+
+
+/**
+ * @brief Perform a CCA and send the given packet if medium is available
+ *
+ * @param[in] device    netdev device, needs to be already initialized
+ * @param[in] data      pointer to the data in the packet buffer
+ *
+ * @return              the return value of device driver's @c send_data()
+ *                      function if medium was avilable
+ * @return              -EBUSY if radio medium was not available
+ *                      to send the given data
+ */
+static int send_if_cca(gnrc_netdev_t *device, gnrc_pktsnip_t *data)
+{
+    netopt_enable_t hwfeat;
+
+    /* perform a CCA */
+    DEBUG("Checking radio medium availability...\n");
+    int res = device->driver->get(dev,
+                                  NETOPT_IS_CHANNEL_CLR,
+                                  (void *) &hwfeat,
+                                  sizeof(netopt_enable_t));
+    if (res < 0) {
+        /* normally impossible: we got a big internal problem! */
+        DEBUG("!!! DEVICE DRIVER FAILURE! TRANSMISSION ABORTED!\n");
+        return -ECANCELED;
+    }
+
+    /* if medium is clear, send the packet and return */
+    if (hwfeat == NETOPT_ENABLE) {
+        DEBUG("Radio medium available: sending packet.\n");
+        return dev->driver->send_data(device, data);
+    }
+
+    /* if we arrive here, medium was not available for transmission */
+    DEBUG("Radio medium busy.\n");
+    return -EBUSY;
+}
+
+/*------------------------- "EXPORTED" FUNCTIONS -------------------------*/
+
+void set_csma_mac_min_be(uint8_t val)
+{
+    macMinBE = val;
+}
+
+void set_csma_mac_max_be(uint8_t val)
+{
+    macMaxBE = val;
+}
+
+void set_csma_mac_max_csma_backoffs(uint8_t val);
+{
+    macMaxCSMABackoffs = val;
+}
+
+
+int csma_ca_send(gnrc_netdev_t *dev, gnrc_pktsnip_t *pkt);
+{
+    netopt_enable_t hwfeat;
+
+    /* Does the transceiver do automatic CSMA/CA when sending? */
+    int res = dev->driver->get(dev,
+                               NETOPT_CSMA,
+                               (void *) &hwfeat,
+                               sizeof(netopt_enable_t));
+    bool ok = false;
+    switch (res) {
+    case -ENODEV:
+        /* invalid device pointer given */
+        return -ENODEV;
+    case -ENOTSUP:
+        /* device doesn't make auto-CSMA/CA */
+        ok = false;
+        break;
+    case -EOVERFLOW:  /* (normally impossible...*/
+    case -ECANCELED:
+        DEBUG("!!! DEVICE DRIVER FAILURE! TRANSMISSION ABORTED!\n");
+        /* internal driver error! */
+        return -ECANCELED;
+    default:
+        ok = (hwfeat == NETOPT_ENABLE);
+    }
+
+    if (ok) {
+        /* device does CSMA/CA all by itself: let it do its job */
+        DEBUG("Network device does hardware CSMA/CA\n");
+        return dev->driver->send_data(dev, pktsnip);
+    }
+
+    /* if we arrive here, then we must perform the CSMA/CA procedure
+       ourselves by software */
+    genrand_init(xtimer_now());
```

Comment:
Just did a little research. Currently it isn't seeded by auto_init (see #3425). But calling it every time introduces quite some overhead, see [the source](https://github.com/RIOT-OS/RIOT/blob/86da62897183f11e4754dfee2011fbc09f3f863a/sys/random/mersenne.c#L58-L70). I guess you are right that it doesn't hurt "randomness-wise", but it doesn't really get better either if the seed was good in the first place.